libs/vre/shared/app-representations/src/lib/still-image/still-image.component.ts

```diff
@@ -1,616 +1,107 @@
-      return this.resource.properties[Constants.HasStillImageFileValue][0] as ReadStillImageExternalFileValue;
-    } else return undefined;
-  }
-
-  isPng: boolean = false;
-
-  imageFileInfo: FileInfo | undefined = undefined;
-
-  get isReadStillImageExternalFileValue(): boolean {
-    return !!this.imageFileValue && this.imageFileValue.type === Constants.StillImageExternalFileValue;
-  }
-
-  get userCanView() {
-    return this.imageFileValue && ResourceUtil.userCanView(this.imageFileValue);
-  }
+  @ViewChild('osdViewer') osdViewerElement!: ElementRef;
 
-  get usercanEdit() {
-    return ResourceUtil.userCanEdit(this.resource);
-  }
-
-  failedToLoad = false;
-
-  imageFormatIsPng = this._resourceFetcherService.settings.imageFormatIsPng;
-  regionDrawMode = false; // stores whether viewer is currently drawing a region
-  private _regionDragInfo; // stores the information of the first click for drawing a region
-  private _viewer: OpenSeadragon.Viewer | undefined;
-  private _regions: PolygonsForRegion = {};
+  isViewInitialized = false;
+  isPng = false;
 
   constructor(
-    @Inject(DspApiConnectionToken)
-    private _dspApiConnection: KnoraApiConnection,
-    private _dialog: MatDialog,
-    private _domSanitizer: DomSanitizer,
-    private _elementRef: ElementRef,
-    private _matIconRegistry: MatIconRegistry,
-    private _notification: NotificationService,
-    private _renderer: Renderer2,
-    private _rs: RepresentationService,
-    private _regionService: RegionService,
-    private _resourceFetcherService: ResourceFetcherService,
-    private _cd: ChangeDetectorRef
-  ) {
-    OpenSeadragon.setString('Tooltips.Home', '');
-    OpenSeadragon.setString('Tooltips.ZoomIn', '');
-    OpenSeadragon.setString('Tooltips.ZoomOut', '');
-    OpenSeadragon.setString('Tooltips.FullPage', '');
-
-    // own draw region icon; because it does not exist in the material icons
-    this._matIconRegistry.addSvgIcon(
-      'draw_region_icon',
-      this._domSanitizer.bypassSecurityTrustResourceUrl('/assets/images/draw-region-icon.svg')
-    );
-  }
-
-  ngOnInit() {
-    this._setupViewer();
-    this._loadImages();
-
-    this._regionService.imageIsLoaded$
-      .pipe(
-        filter(loaded => loaded),
-        switchMap(() =>
-          combineLatest([this._regionService.showRegions$.pipe(distinctUntilChanged()), this._regionService.regions$])
-        )
-      )
-      .pipe(filter(() => !!this._viewer)) // only proceed if the viewer is already set up
-      .subscribe(([showRegion, regions]) => {
-        this._removeOverlays();
-
-        if (showRegion) {
-          this._renderRegions();
-        }
-      });
-
-    this._regionService.highlightRegion$.subscribe(region => {
-      if (region === null) {
-        this._unhighlightAllRegions();
-        return;
-      }
-      this._highlightRegion(region);
+    public osdDrawerService: OsdDrawerService,
```

Comment:
API Injection token is missing

Reply:
what do you mean ? OsdDrawerService is injected in the component (see line 40)

Reply:
My IDE was complaining that it is missing in the still-image component constructor as well, but apparantly it is not required